sys/arm64/arm64/exception_morello.S

```diff
@@ -134,7 +134,7 @@ __FBSDID("$FreeBSD$");
 	ldp	c26, c27, [sp, #(TF_X + 26 * 16)]
 	ldp	c28, c29, [sp, #(TF_X + 28 * 16)]
 .else
-	ldr	     x29, [sp, #(TF_X + 29 * 16)]
+	ldr	     c29, [sp, #(TF_X + 29 * 16)]
```

Comment:
Yeah this was wrong but I'm surprised this ever matters in practice. We're not using clr anywhere that I know of. Is the compiler ending up using it occasionally as a temporary register and you took an interrupt at an inopportune time?

Reply:
I noticed it when looking at the disassembly of the exception handlers and seeing it load an `x` register looked out of place. I don't expect it will affect anything until we move to a purecap kernel.

Reply:
I think the else case needs to be removed and we just use the el0 code. Any use of c18-c29 with a tag set could be problematic if an interrupt is triggered between loading and using these registers.

Reply:
Guard the `.if` and `.else ... .endif` with `#ifdef __CHERI_PURE_CAPABILITY__` then? The optimisation works for a purecap kernel as the full capabilities are saved and restored naturally as part of the ABI.